src/components/ImageView/index.js

```diff
@@ -120,27 +117,36 @@ class ImageView extends PureComponent {
     }
 
     /**
-     * When open image, set image width, height and zoomScale.
+     * When open image, set image width, height.
      * @param {Number} imageWidth
      * @param {Number} imageHeight
      */
     setImageRegion(imageWidth, imageHeight) {
         if (imageHeight <= 0) {
             return;
         }
-        const containerHeight = this.state.containerHeight;
-        const containerWidth = this.state.containerWidth;
-        const width = imageWidth;
-        const height = imageHeight;
-        const newZoomScale = Math.min(containerWidth / width, containerHeight / height);
 
+        this.setScale(this.state.containerWidth, this.state.containerHeight, imageWidth, imageHeight);
         this.setState({
-            imgWidth: width,
-            imgHeight: height,
-            zoomScale: newZoomScale,
+            imgWidth: imageWidth,
+            imgHeight: imageHeight,
         });
     }
 
+    /**
+     * @param {Number} containerWidth
+     * @param {Number} containerHeight
+     * @param {Number} imageWidth
+     * @param {Number} imageHeight
+     */
+    setScale(containerWidth, containerHeight, imageWidth, imageHeight) {
+        if (!containerWidth || !imageWidth) {
```

Comment:
Why is this here?

Reply:
We have `onContainerLayoutChanged` and `setImageRegion` functions that were called asynchronously. However, in only one of them, the `zoomScale` information could be calculated exactly. For this reason, we wrote this method to avoid code duplication.

Reply:
Sorry, I am asking why this if statement is here? Why are we only checking `*Width` and not `*Height`?

Reply:
@metehanozyurt let's keep the discussion about this here. You said:

>I think the width is enough for us to understand that the photo has been loaded.

Why should we NOT be checking the height? It is not immediately obvious to a newcomer / outsider to the code (eg, me)

Reply:
All photos have width and height values. In fact, we check whether the installation is complete, and if there has been a change in both, we perform the calculation. The value of both is initially 0. We just chose between width and height. We could have included height, but then it would be unnecessary control @flodnv .

Reply:
You're right @flodnv , I added the heights. I added it at the end to avoid extra checking. thank you for the explanation.